pudb/__init__.py

```diff
@@ -132,7 +138,14 @@ def runscript(mainpyfile, args=None, pre_run="", steal_output=False):
     import sys
     if args is not None:
         prev_sys_argv = sys.argv[:]
-        sys.argv = [mainpyfile] + args
+        if run_as_module:
+            sys.argv = args
+        else:
+            sys.argv = [mainpyfile] + args
+    else:
+        if run_as_module:
+            prev_sys_argv = sys.argv[:]
+            sys.argv = sys.argv[2:]
```

Comment:
This is not necessarily correct. `OptionParser` as called below will tolerate `-m SOMETHING` in places other than the first two arguments.

Reply:
Hm, I admit I've added it without fully understanding...
But when I do `pudb3 -m json.tool`, it tries to open `json.tool` if I don't do this as if it were the JSON file.
I'll do some more tests.

Reply:
Ahh, you're right, I did a few more testing locally and I got it now, I actually need to update `sys.argv` before calling runmodule as it's done for runscript -- will fix in a jiffy!